src/Microsoft.AspNetCore.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -229,7 +229,7 @@ public void Copy(Operation operation, object objectToApplyTo)
             if (TryGetValue(operation.from, objectToApplyTo, operation, out var propertyValue))
             {
                 // Create deep copy
-                var copyResult = ConversionResultProvider.CopyTo(propertyValue, propertyValue.GetType());
+                var copyResult = ConversionResultProvider.CopyTo(propertyValue, propertyValue?.GetType());
```

Comment:
Wouldn't this null ref: https://github.com/aspnet/JsonPatch/blob/1a1b34194b387175ef1bc12d2af6480ae1ba9c0c/src/Microsoft.AspNetCore.JsonPatch/Internal/ConversionResultProvider.cs#L18-L22

Reply:
@pranavkm 
In the described scenario, no. If you check the only usage "CopyTo" method, typeToConvertTo will be null only in case of value == null and this statement will never execute. In case of "ConvertTo" method typeToConvertTo always populated from Path segment and should have value in any case.